eng/release/DiagnosticsReleaseTool/Common/AzureBlobPublisher.cs

```diff
@@ -0,0 +1,267 @@
+        }
+
+        private async Task<BlobContainerClient> GetClient(CancellationToken ct)
+        {
+            if (_client == null)
+            {
+                BlobServiceClient serviceClient = new BlobServiceClient(AccountBlobUri, AccountCredential, BlobOptions);
+                _logger.LogInformation($"Attempting to connect to {serviceClient.Uri} to store blobs.");
+
+                BlobContainerClient newClient;
+                int attemptCt = 0;
+                do
+                {
+                    try
+                    {
+                        newClient = (await serviceClient.CreateBlobContainerAsync(_containerName, PublicAccessType.None, metadata: null, ct)).Value;
+                        _logger.LogInformation($"Created new contianer: {_containerName}.");
+                    }
+                    // 409 is the error code for an existing blob container already existing
+                    catch (RequestFailedException rfe) when (rfe.Status == 409)
+                    {
+                        _logger.LogInformation($"Found existing contianer: {_containerName}.");
+                        // Got an existing container
+                        newClient = serviceClient.GetBlobContainerClient(_containerName);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogWarning(ex, $"Failed to create or access {_containerName}, retrying with new name.");
+                        continue;
+                    }
+
+                    try
+                    {
+                        // Add the new (or update existing) "download" policy to the container
+                        // This is used to mint the SAS tokens without an expiration policy
+                        // Expiration can be added later by modifying this policy
+                        BlobSignedIdentifier downloadPolicyIdentifier = new BlobSignedIdentifier()
+                        {
+                            Id = AccessPolicyDownloadId,
+                            AccessPolicy = new BlobAccessPolicy()
+                            {
+                                Permissions = "r",
+                                PolicyStartsOn = new DateTimeOffset(DateTime.UtcNow),
+                                PolicyExpiresOn = new DateTimeOffset(DateTime.UtcNow.AddDays(_sasValidDays)),
+                            }
+                        };
+                        _logger.LogInformation($"Writing download access policy: {AccessPolicyDownloadId} to {_containerName}.");
+                        await newClient.SetAccessPolicyAsync(PublicAccessType.None, new BlobSignedIdentifier[] { downloadPolicyIdentifier }, cancellationToken: ct);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogWarning(ex, $"Failed to write access policy for {_containerName}, retrying with new name.");
```

Comment:
nit: This implies that a new name will be generated, but I don't see that behavior. It looks like it'll try again with the same container name.